src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -238,26 +238,25 @@ internal unsafe int IndexOfCore(string source, string target, int startIndex, in
             Debug.Assert((options & CompareOptions.OrdinalIgnoreCase) == 0);
             Debug.Assert((options & CompareOptions.Ordinal) == 0);
 
-#if CORECLR
-            if (_isAsciiEqualityOrdinal && CanUseAsciiOrdinalForOptions(options) && source.IsFastSort() && target.IsFastSort())
+            int index;
+
+            if (_isAsciiEqualityOrdinal && CanUseAsciiOrdinalForOptions(options))
             {
```

Comment:
could you please try the case of string has nul characters and using ignore nonspace option. Just to ensure no regression is encountered?

Reply:
I have tried:
```
using System;
using System.Globalization;

class Program
{
    static void Main(string[] args)
    {
        string s = "a\u0000b";
        CompareInfo ci = CultureInfo.CurrentCulture.CompareInfo;

        Console.WriteLine(ci.IndexOf(s, "a", CompareOptions.IgnoreNonSpace));
        Console.WriteLine(ci.IndexOf(s, "\u0000", CompareOptions.IgnoreNonSpace));
        Console.WriteLine(ci.IndexOf(s, "b", CompareOptions.IgnoreNonSpace));
        Console.WriteLine(ci.IndexOf(s, "c", CompareOptions.IgnoreNonSpace));
    }
}
```

It prints 0, 0, 2, -1 both before and after the change.

Reply:
Thanks @jkotas for confirming.